build_tools/cmake/iree_cc_binary.cmake

```
@@ -134,8 +134,17 @@ function(iree_cc_binary)
   set_property(TARGET ${_NAME} PROPERTY CXX_STANDARD ${IREE_CXX_STANDARD})
   set_property(TARGET ${_NAME} PROPERTY CXX_STANDARD_REQUIRED ON)
 
-  install(TARGETS ${_NAME}
-          RENAME ${_RULE_NAME}
-          COMPONENT ${_RULE_NAME}
-          RUNTIME DESTINATION bin)
+  if (${_RULE_EXCLUDE_FROM_ALL})
```

nit: no space between `if` and `(`, and I think you can omit the `${}` here (copy how `_RULE_TESTONLY` is used above)
```suggestion
  if(_RULE_EXCLUDE_FROM_ALL)
```

https://cmake.org/cmake/help/latest/command/if.html#basic-expressions

> `if(<variable>)`
True if given a variable that is defined to a value that is not a false constant. False otherwise, including if the variable is undefined